src/intersections/lines.jl

```diff
@@ -43,14 +28,47 @@ end
 function intersectpoint(l1::Line, l2::Line)
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
+  λ₁, _ = intersectparameters(a, b, c, d)
+  a + λ₁ * (b - a)
+ end
+
+#= 
+Compute the parameters of the lines defined by the points a -> b and c -> d 
+so that the resulting points have minimal distance.
+
+The ranks help to identify the different types of intersection:
 
-  v1  = a - b
-  v2  = c - d
-  v12 = a - c
+Intersection: r == rₐ == 2
+Collinear: r == rₐ == 1
+No intersection: r != rₐ
+  No intersection and parallel:  r == 1, rₐ == 2
+  No intersection, skew lines: r == 2, rₐ == 3
+=#
```

Comment:
Thank you @dorn-gerhard for checking all the comments and reviews here carefully. Keep in mind that the more comments and explanations you share upfront, the easier it will be to merge delicate fundamental changes to the master branch.

I still feel that this introductory comment is not self-contained and that anyone reading it for the first time will get lost. Can you try to rewrite it telling the full story upfront? Defining the variables, the matrices involved in the algorithm, etc? After you help the reader with this introductory comment, the function body can just contain small important explanations not already explained in the introductory comment.

I think that after that we will finally be able to squash and merge. Thanks for pushing it further, I am doing my best to review it in the middle of other activities.